### DISCUSSION TITLE
SmartAutoClicker-1.5.0

### DISCUSSION THREAD

**Discussion by Nain57** [Nain57/Smart-AutoClicker#133] [Announcements]
# New Feature:

## Import/Export scenario
You can now create an archive from one or many scenario and restore it later. Please note that the scenario are not meant to work on all devices, and one created on a small screen will not work on a big screen.

# Bug fixes:

## Overlay can't move on some part of the screen.
This bug is recurring every time I try to fix it for another device. Now concrete tests cases have been written for all know occurrence of this issue, and they should be no more regressions on it.

## Colors not differentiated during detection
The detection algorithm was unable to differentiate the same shape with different colors. Now it is.
 

<hr /><em>This discussion was created from the release <a href='https://github.com/Nain57/Smart-AutoClicker/releases/tag/1.5.0'>SmartAutoClicker-1.5.0</a>.</em>

**Comment by zakius** [comment]
> The detection algorithm was unable to differentiate the same shape with different colors. Now it is.

that explains a lot! thanks for the fix

**Reply by aloysiuscalvin** [reply]
Currently im using this app for Play Together fishing -> detecting exclamation mark with different color

how much tolerate percent it is needed to detect the exclamation mark with different color?

in 1.4 I usually made my screen monochrome so that the color difference will not affect too much, but it was a pain to continuously change my phone to monochrome every time I want to grind

I already done several test with the percentage but i couldn't figure out the perfect percentage to fully utilize the algorithm to differentiate the same shape with different colors

**Reply by Nain57** [reply]
It will really depends on your case.
The more shapes there is on the condition, the easiest the matching will be.

To help you find the correct tolerance, you can use the debug mode (long press on the play button). It will display you the current confidence rate, even when nothing is detected.
For example, if your condition is shown on screen and not detected, and the displayed confidence rate is 92%, you must have at least 8% in tolerance to detect it correctly.
